homeassistant/components/amcrest.py

```diff
@@ -93,22 +94,23 @@ def setup(hass, config):
     amcrest_cams = config[DOMAIN]
 
     for device in amcrest_cams:
-        camera = AmcrestCamera(device.get(CONF_HOST),
-                               device.get(CONF_PORT),
-                               device.get(CONF_USERNAME),
-                               device.get(CONF_PASSWORD)).camera
         try:
+            camera = AmcrestCamera(device.get(CONF_HOST),
```

Comment:
Why move this inside the try? Does it also make a connection?

Reply:
And if it does, do we still need `camera.current_time` statement below?

Reply:
yes it makes a connection.  I didn't write the original code and have no idea if the ```camera.current_time``` is still needed.  I do know that the connetion error thrown by the initialization is different from that from current_time though.

Reply:
the constructor does setup the connection to the camera and completes the authorzation, so I THINK it should be sufficient.  ```current_time``` sends a command to the camera and verifies the response, so I suppose it could catch some corner case of a camera that doesn't support the expected amcrest command interface.  I don't know the history of this component, but if the only point was to trigger the connection, it shouldn't be needed.

Reply:
Alright, let's keep it in .